src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -65,8 +65,9 @@ public RpcInitializationService(IOptionsMonitor<ScriptApplicationHostOptions> ap
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
-            if (Utility.CheckAppOffline(_applicationHostOptions.CurrentValue.ScriptPath))
+            if (Utility.CheckAppOffline(_environment, _applicationHostOptions.CurrentValue.ScriptPath))
             {
+                _logger.LogDebug("AppOffline in RpcInitializationService");
```

Comment:
nit: "App is offline. Skipping Rpc initialization"

Reply:
Fixed.